From 9dfd41f95cc6f5d7e26a121f31eaa0e589c580bb Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 28 Jan 2015 00:41:03 -0500
Subject: [PATCH] add Setter for player name in AsyncPlayerPrelogin

This is so we can change the players name early in login process
---
 .../java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
index 6c09ea6c6..54a51ea82 100644
--- a/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
+++ b/src/main/java/org/bukkit/event/player/AsyncPlayerPreLoginEvent.java
@@ -18,7 +18,7 @@ public class AsyncPlayerPreLoginEvent extends Event {
     private static final HandlerList handlers = new HandlerList();
     private Result result;
     private String message;
-    private final String name;
+    private String name; // EMC
     private final InetAddress ipAddress;
     private final UUID uniqueId;
 
@@ -170,6 +170,7 @@ public class AsyncPlayerPreLoginEvent extends Event {
     public String getName() {
         return name;
     }
+    public void setName(String name) { this.name = name; } // EMC
 
     /**
      * Gets the player IP address.
-- 
2.21.0

